src/crypto/tls/ech.go

```diff
@@ -643,8 +670,8 @@ func echDecodeClientHelloInner(encodedData, outerData, outerSessionId []byte) []
 					}
 					handledOuterExtensions = true
 
-					// Read the set of outer extension code points.
-					outer := make(map[uint16]bool)
+					// Read the referenced outer extensions.
+					referencedExts := make([]uint16, 0, 10)
```

Comment:
The max will always be `10`?

Reply:
No, it's possible for there to be more, but we don't know how many at this point. Allocating enough space for `10` initially seemed like a a sensible prediction. If we go over, the runtime will just allocate more memory as needed.